src/main/webapp/script.js

```diff
@@ -90,18 +98,25 @@ async function generateGraph() {
  * Ensures that the depth is an integer between 0 and 20
  */
 function getUrl() {
-  let selectedDepth = document.getElementById('num-layers').value;
-  if (selectedDepth.length === 0) {
+  const depthElem = document.getElementById('num-layers');
+  let selectedDepth = 0;
```

Comment:
is there a reason you needed to define this variable?

Reply:
Just in case the num-layers input field isn't defined so that I can check if it's null before trying to get its value. This came up in one of my tests where I have to set up the document by hand and forgot to initialize this field so I was getting an error. Do you think I should leave the check or modify my tests to comply?

Reply:
oh it's not a big deal haha should be good the way it is!